src/phoenix/experimental/evals/models/base.py

```diff
@@ -30,35 +31,56 @@
 )
 
 
-def create_base_retry_decorator(
-    error_types: List[Type[BaseException]],
-    min_seconds: int,
-    max_seconds: int,
-    max_retries: int,
-) -> Callable[[Any], Any]:
-    """Create a retry decorator for a given LLM and provided list of error types."""
-
-    # TODO: Nice logging. The logging implemented is huge and overwhelming
-    _logging = before_sleep_log(logger, logging.WARNING)
-
-    def _before_sleep(retry_state: RetryCallState) -> None:
-        _logging(retry_state)
-        return None
-
-    retry_instance: retry_base = retry_if_exception_type(error_types[0])
-    for error in error_types[1:]:
-        retry_instance = retry_instance | retry_if_exception_type(error)
-    return retry(
-        reraise=True,
-        stop=stop_after_attempt(max_retries),
-        wait=wait_random_exponential(multiplier=1, min=min_seconds, max=max_seconds),
-        retry=retry_instance,
-        # before_sleep=_before_sleep,
-    )
+@contextmanager
+def set_verbosity(
+    model: "BaseEvalModel", verbose: bool = False
+) -> Generator["BaseEvalModel", None, None]:
+    try:
+        model._verbose = verbose
+        yield model
+    finally:
+        model._verbose = False
```

Comment:
Suggestion: Instead of changing a private property on an instance, how about a `verbose` property that can be set via a setter and getter?

Reply:
resolved offline